packages/validator/src/services/validatorStore.ts

```diff
@@ -299,17 +349,35 @@ export class ValidatorStore {
     const domain = this.config.getDomain(contribution.slot, DOMAIN_CONTRIBUTION_AND_PROOF);
     const signingRoot = computeSigningRoot(ssz.altair.ContributionAndProof, contributionAndProof, domain);
 
+    const signableMessage: SignableMessage = {
+      singablePayload: {
+        type: "SYNC_COMMITTEE_CONTRIBUTION_AND_PROOF",
+        data: contributionAndProof,
+      },
+      forkInfo: this.getForkInfo(contributionAndProof.contribution.slot),
+    };
+
     return {
       message: contributionAndProof,
-      signature: await this.getSignature(duty.pubkey, signingRoot),
+      signature: await this.getSignature(duty.pubkey, signingRoot, signableMessage),
     };
   }
 
   async signAttestationSelectionProof(pubkey: BLSPubkeyMaybeHex, slot: Slot): Promise<BLSSignature> {
     const domain = this.config.getDomain(slot, DOMAIN_SELECTION_PROOF);
     const signingRoot = computeSigningRoot(ssz.Slot, slot, domain);
 
-    return await this.getSignature(pubkey, signingRoot);
+    const signableMessage: SignableMessage = {
```

Comment:
If the web3signer is not enabled, all this objects are being created for nothing right?

Reply:
Updated to only create the Singable messages when web3signer is enabled.